maven-api-impl/src/main/java/org/apache/maven/internal/impl/model/DefaultModelBuilder.java

```diff
@@ -591,13 +593,14 @@ Model transformFileToRaw(Model model) {
                     }
                     if (depBuilder != null) {
                         newDeps.add(depBuilder.build());
-                        modified = true;
                     } else {
                         newDeps.add(dep);
                     }
+                } else {
+                    newDeps.add(dep);
                 }
             }
-            return modified ? model.withDependencies(newDeps) : model;
+            return model.withDependencies(newDeps);
```

Comment:
`model.withDependencies()` will always create a new object if not modified.
I don't see why this would be needed.